android-test/src/androidTest/java/okhttp/android/test/StrictModeTest.kt

```diff
@@ -0,0 +1,69 @@
+package okhttp.android.test
+
+import android.os.StrictMode
+import android.os.StrictMode.ThreadPolicy
+import android.os.strictmode.Violation
+import assertk.assertThat
+import assertk.assertions.hasSize
+import assertk.assertions.isEmpty
+import assertk.assertions.isEqualTo
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.internal.platform.Platform
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.parallel.Isolated
+
+@Isolated
+class StrictModeTest {
+  private val violations = mutableListOf<Violation>()
+
+  @AfterEach
+  fun cleanup() {
+    StrictMode.setThreadPolicy(
+      ThreadPolicy.Builder()
+        .permitAll()
+        .build(),
+    )
+  }
+
+  @Test
+  fun testInit() {
+    Platform.resetForTests()
+
+    applyStrictMode()
+
+    // Not currently safe
+    // See https://github.com/square/okhttp/pull/8248
+    OkHttpClient()
+
+    assertThat(violations).hasSize(1)
+    assertThat(violations[0].message).isEqualTo("newSSLContext")
+  }
+
+  @Test
+  fun testNewCall() {
+    Platform.resetForTests()
+
+    val client = OkHttpClient()
+
+    applyStrictMode()
+
+    // Safe on main
+    client.newCall(Request("https://google.com/robots.txt".toHttpUrl()))
+
+    assertThat(violations).isEmpty()
+  }
+
+  private fun applyStrictMode() {
+    StrictMode.setThreadPolicy(
+      ThreadPolicy.Builder()
+        .detectCustomSlowCalls()
+        .penaltyListener({ it.run() }) {
+          violations.add(it)
+        }
```

Comment:
Hmmm. When was this added? I would assume after our minSdk, so it should really have an `@SdkSuppress(minSdk=something)` on the class.

Reply:
I'll check, also why it's not failing if so.

Reply:
Ahhh, just the tests. And not running because of https://github.com/mannodermaus/android-junit5/issues/225 and a minSdk 26 requirement for tests.

Reply:
Ah. JUnit 5 continues to prove that it's the wrong choice for any project.